### ISSUE TITLE
[Nominee task 1 of 3] Agree and Sign ETO Agreements

### ISSUE THREAD

**Issue by andywalton** [Neufund/platform-frontend#3237]
This ticket is part of the Nominee Dashboard epic #3203

**TL;DR**

We're building a new dashboard to allow our new user type Nominee to register/login, connect with an Issuer and thereafter to perform tasks required of them before, during and after an Issuers fundraising campaign.

**Tasks for ticket**
At this stage a nominee has a successfully linked themselves to an Issuer.

The Nominee will at this stage have nothing to do until they are told to complete their first Nominee task: accepting and signing the ETO Agreements

There are two agreements that needs to be signed in following order
* Token Holder Agreement (THA). 
* Reservation and Acquisition Agreement (RAAA)

Steps are identical for both. You execute RAAA signing when you see that THA is already signed (see technical details below)

The steps are as follows:

- [ ] We need to tell them they should do this (a card on the dashboard with a call to action)
- [ ] This will trigger a modal that will allow them to "Agree" - download and view this THA after
- [ ] They will then need to Agree via clicking a checkbox (which will enable the modal call to action allowing them Sign
- [ ] Signing will move the modal to the next step - ie. to Sign via entering their Light Wallet password.
- [ ] This will results in an on-chain transaction and we should showcase that this is happening using our transaction status flow and a summary screen that shows it's either in progress, has an error or is still pending.
- [ ] if THA is signed but RAAA is not signed you repeat steps above for RAAA

**Designs**
[See designs in Zeplin](https://app.zeplin.io/project/5d2b7e0bd1281e565d9d08e4)

Scroll down to section: 5) Nominee task #1 - Sign THA
RAAA is identical, you just need to download different template (as pdf) and execute transaction agains different contract.

**Acceptance**
Is as per the steps above - we notify the Nominee that they should sign the THA, the can view and download this THA, agree that they are happy with it and sign for this using our transaction signer. We notify them of the success of this.

**Notes:**
We aren't handling the case where they might not agree to the THA, this is for a later phase and should be handled via support or contacting an account manager. For now, they need to do it. :)

Also, if the THA changes, it will ask the Nominee to redo this step but for the new THA.

**Comment by rudolfix** [comment]
here's tech information for THA signing and later steps. you can copy relevant part to tickets once they are available.

@andywalton THA situation is different - Nominee is an owner of THA. he can sign and update THA as s/he wants and then it's up to issuer to accept the final terms. This happens when setting the date.

My suggestion is to ignore this problem, however we should have a step in issuer dashboard that
1. you are waiting for Nominee to sign THA (do we have it)
2. that Nominee signed THA and you are ready to set the date of the ETO and start it (this one we already have)

**THA**
- eto `state` is `on_chain` (we have contracts on the network - obviously)
- you must obtain Agreement associated with equityToken. it exposes `IAgreement` interface. I suggest to read comments in https://github.com/Neufund/platform-contracts/blob/master/contracts/Standards/IAgreement.sol
- check if there are agreements by `amendmentsCount()`
- if so get last agreement by pastAgreement(uint256 amendmentIndex)
- if ipfs_hash returned is same as hash that is stored in `templates` from `company_token_holder_agreement_template` then correct template is already signed in and step is done
- if there are no agreements or agreement ipfs_hash is different you must use `amendAgreement(string agreementUri)` where you provide ipfs hash prefixed with `ipfs:`
- equityToken address is part of `eto_data` returned from backend - starting from the on-chain state

**withdrawing capital increase**
- nominee will get capital increase on transition to Signing state. this must be paid into Issuer account in EUR and shown to the notary. it's an offchain process.
- obtain capital increase by calling `contributionSummary`, second field is the value in nEUR
- as long as nEUR wallet balance is >= capital increase (mind rounding) you should show a reminder widget that can lead to REDEEM with correct amount already set (and non editable)

**Confirming Signed ISHA**
The task of this step is to prove to ETO smart contract that both parties signed the same investment agreement. 
- eto must be in Signing eto smart contracts state
- First issuer uploads signed ISHA and adds ipfs hash via calling `companySignsInvestmentAgreement`
when 
- nominee checks if `signedInvestmentAgreementUrl()` returns anything. we do not show this value to the user! see later
- Nominee and Issuer must share same binary version of signed ISHA - preferably sent by notary
- Nominee "uploads" ISHA. We calculate IPFS hash. this should happen purely on frontend!
https://ethereum.stackexchange.com/questions/44506/ipfs-hash-algorithm
- you compare this with what is in `signedInvestmentAgreementUrl()` if there's any difference we show an error!
- if no differences you call `nomineeConfirmsInvestmentAgreement`, if transaction is completed eto will transition to Claim state!

**Comment by glupiawymowka** [comment]
@rudolfix I guess I have no idea how to test it. 
I went through all the nominee fixtures and only saw "you're all up to date" modal. 
Then, I took the nominee and issuer SIGNING fixtures, signed ISHA as the issuer - no change on the nominee's side. 
Either I'm doing something wrong or we don't have the necessary fixture OR something is wrong here. 
Could we go through this ticket together?

**Comment by glupiawymowka** [comment]
tested on stage
